PowerEditor/src/WinControls/Preference/preferenceDlg.cpp

```diff
@@ -5596,65 +5611,100 @@ intptr_t CALLBACK SearchingSubDlg::run_dlgProc(UINT message, WPARAM wParam, LPAR
 
 		case WM_COMMAND:
 		{
-			switch (wParam)
+			if (LOWORD(wParam) == IDC_INSELECTION_THRESHOLD_EDIT)
 			{
-				case IDC_CHECK_FILL_FIND_FIELD_WITH_SELECTED:
+				if (HIWORD(wParam) == EN_KILLFOCUS)

```

Comment:
@donho & @alankilborn,

I haven't tried it with `EN_CHANGE`.
But I'm wondering if it's `EN_KILLFOCUS` vs. `EN_CHANGE` or another variant of https://github.com/notepad-plus-plus/notepad-plus-plus/issues/14108#issuecomment-1719653099.

Reply:
In order to create a numeric edit box control in the preferences, I went looking for an existing one to model upon.  I found the ones in _Recent Files History_:

![image](https://github.com/notepad-plus-plus/notepad-plus-plus/assets/77065706/12d7a25e-9266-44f7-a9ba-76d642409e5a)

These use `EN_KILLFOCUS`, hence my usage in currrent PR.  I will change current PR, but I'd think that `EN_KILLFOCUS` usage for  _Recent Files History_ should also change -- should I create an issue for that?

Reply:
https://github.com/notepad-plus-plus/notepad-plus-plus/pull/13245#issuecomment-1450230191.

Reply:
@Yaron10, there wasn't much explanation there, but I think I understand.  In that case, as soon as the `3` is processed (by EN_CHANGE) the list of recent files is truncated to be of length 3, so that when the `0` is added and 30 is processed (by the following EN_CHANGE), there isn't any data remaining with which to re-lengthen the list.

Reply:
@alankilborn,

Very well explained.
It was easier to understand when that binary (using `EN_CHANGE`) was available.

Reply:
> but I'd think that EN_KILLFOCUS usage for Recent Files History should also change -- should I create an issue for that?

@alankilborn 
If you can reproduce an issue, then please do it.

Reply:
@donho : Discussion with @Yaron10 revealed no changes needed to Recent Files History section.

Changed the code in this PR to use EN_CHANGE.